### PR TITLE
Regex solver locations

### PR DESCRIPTION
**Pull request by laurentheirendt**
Auto-detection of solvers based on default installation folder locations independent of version number.

**I hereby confirm that I have:**

- [x] Tested my code on my own machine
- [x] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [x] Selected `develop` as a target branch (top left drop-down menu)

### CODE REVIEW

File: src/base/solvers/configEnvVars.m [hunk 1]
```diff
@@ -20,75 +20,77 @@ function configEnvVars(printLevel)
...
-        global MOSEK_PATH;
+
+        % initialize global variables
         GUROBI_PATH = []; ILOG_CPLEX_PATH = []; TOMLAB_PATH = []; MOSEK_PATH = [];
 
+        % define default locations of the root installation folder
+        defaultLocationsUNIX = {'/Applications/', ...
+                                '~/Applications/', ...
+                                '/opt/', ...
+                                '/', ...
```
Comment (syarra) [inline src/base/solvers/configEnvVars.m]: Who does install a solver on `/`? This is really risky!

Comment (laurentheirendt) [inline src/base/solvers/configEnvVars.m]: Ok, let's remove it then.

File: src/base/solvers/configEnvVars.m [hunk 2]
```diff
@@ -102,16 +103,83 @@ function configEnvVars(printLevel)
...
+                folderPattern = solverPaths{k, 3};
+                folderNameVect = solverPaths{k, 4};
+
+                % define additional subfolders for IBM ILOG CPLEX and MOSEK
+                if k == 1 || k == 4
+
+                    % initialize and empty cell for storing folder names
+                    tmpFolderNameVect = {};
+
+                    % loop through all default location and add the subfolders
```
Comment (syarra) [inline src/base/solvers/configEnvVars.m]: What about this?
````
                    len = length(folderNameVect);
          
                    % loop through all default location and add the subfolders
                    if k == 1 % IBM ILOG CPLEX
                        tmpFolderNameVect(end+1:end+len) = strcat(folderNameVect, ['IBM' filesep 'ilog' filesep]);
                        tmpFolderNameVect(end+1:end+len) = strcat(folderNameVect, ['IBM' filesep 'ILOG' filesep]);
                        tmpFolderNameVect(end+1:end+len) = strcat(folderNameVect, ['ibm' filesep 'ilog' filesep]);
                        tmpFolderNameVect(end+1:end+len) = strcat(folderNameVect, ['ibm' filesep 'ILOG' filesep]);
                    end
                    if k == 4 % MOSEK
                        tmpFolderNameVect(end+1:end+len) = strcat(folderNameVect, ['mosek' filesep]);
                        tmpFolderNameVect(end+1:end+len) = strcat(folderNameVect, ['Mosek' filesep]);
                        folderPattern = '';
                    end
````

Comment (laurentheirendt) [inline src/base/solvers/configEnvVars.m]: What about it?

Comment (syarra) [inline src/base/solvers/configEnvVars.m]: It will remove this `for` loop

Comment (laurentheirendt) [inline src/base/solvers/configEnvVars.m]: Good idea 👍  I will change that.

**Comment by syarra** [comment]
Nice PR!!!

**Pull request merged**